openlane/state/state.py

```diff
@@ -121,19 +115,21 @@ def __init__(
 
     def __getitem__(self, key: Union[DesignFormat, str]) -> StateElement:
         if isinstance(key, DesignFormat):
-            id: str = key.value.id
+            id: str = key.id
             key = id
+            if key not in self:
```

Comment:
shouldn't this be handled by line 122

Reply:
No. It would return a key not found error. A lot of code relies on this returning `None` if not found.

Maybe we should just fix that code instead.

Reply:
yeah. isn't that dependent on the implementation of `GenericImmutableDict`